fred/cogs/crashes.py

```diff
@@ -1,72 +1,70 @@
+from os.path import split
 from time import strptime
-from typing import AsyncIterator
+from typing import AsyncIterator, IO
+from urllib.parse import urlparse
 
 import nextcord
-import nextcord.ext.commands as commands
+import regex
 from PIL import Image, ImageEnhance, UnidentifiedImageError
+from nextcord import Message
 from pytesseract import image_to_string, TesseractError
 
-import config
-from libraries import createembed
+from .. import config
+from ..libraries import createembed
+from ..libraries.common import FredCog
 
-REGEX_LIMIT: float = 2
+REGEX_LIMIT: float = 6.9
 
 
-async def regex_with_timeout(*args, **kwargs):
-    try:
-        return await asyncio.wait_for(asyncio.to_thread(re.search, *args, **kwargs), REGEX_LIMIT)
-    except asyncio.TimeoutError:
-        raise TimeoutError(
-            f"A regex timed out after {REGEX_LIMIT} seconds! \n"
-            f"pattern: ({args[0]}) \n"
-            f"flags: {kwargs['flags']} \n"
-            f"on text of length {len(args[1])}"
-        )
+def regex_with_timeout(*args, **kwargs):
+    return asyncio.wait_for(asyncio.to_thread(regex.search, *args, **kwargs), REGEX_LIMIT)
 
 
-class Crashes(commands.Cog):
+class Crashes(FredCog):
     vanilla_info_patterns = [
-        re.compile(r"Net CL: (?P<game_version>\d+)"),
-        re.compile(r"Command Line: (?P<cli>.*)"),
-        re.compile(r"Base Directory: (?P<path>.+)"),
-        re.compile(r"Launcher ID: (?P<launcher>\w+)"),
+        regex.compile(r"Net CL: (?P<game_version>\d+)"),
+        regex.compile(r"Command Line: (?P<cli>.*)"),
+        regex.compile(r"Base Directory: (?P<path>.+)"),
+        regex.compile(r"Launcher ID: (?P<launcher>\w+)"),
     ]
 
-    def __init__(self, bot):
-        self.bot = bot
-        self.logger = logging.Logger("CRASH_PARSING")
-
     @staticmethod
     def filter_epic_commandline(cli: str) -> str:
         return " ".join(filter(lambda opt: "auth" not in opt.lower(), cli.split()))
 
     async def parse_factory_game_log(self, text: str) -> dict[str, str | int]:
         self.logger.info("Extracting game info")
         lines = text.splitlines()
-        vanilla_info_search_area = filter(lambda l: re.match("^LogInit", l), lines)
+        vanilla_info_search_area = filter(lambda l: regex.match("^LogInit", l), lines)
 
         info = {}
-        patterns = self.vanilla_info_patterns[:]
+        patterns = self.vanilla_info_patterns[:]  # shallow copy
+        """
+        This godforsaken loop sequentially finds information, 
+        dropping patterns to look for as they are found (until it runs out of patterns).
+        The patterns have named regex captures which the rest of the code knows the names of.
+        - Borketh
+        """
         for line in vanilla_info_search_area:
             if not patterns:
                 break
-            elif match := await regex_with_timeout(patterns[0], line):
+            elif match := regex.search(patterns[0], line):
```

Comment:
This becomes a blocking call. We need async regex everywhere

Reply:
https://github.com/mrabarnett/mrab-regex?tab=readme-ov-file#multithreading
Also, it's one pattern on one line. That shouldn't be a major blocker even if it does.

Reply:
We don't use multiple threads so the GIG unlocking doesn't help us. This pattern probably will never take much time but it'll still block while doing it. And the one below could take longer on bigger files